components/turret.py

```diff
@@ -70,35 +44,31 @@ def setup(self) -> None:
         self.motor.config_kP(0, self.pidP, 10)
         self.motor.config_kI(0, self.pidI, 10)
         self.motor.config_kD(0, self.pidD, 10)
-        self.motor.configAllowableClosedloopError(0, self.CLOSED_LOOP_ERROR, 10)
+        self.motor.configAllowableClosedloopError(0, self.ACCEPTABLE_ERROR_COUNTS, 10)
+        self.scan_increment = math.radians(10.0)
+        self.index_found = False
+        self.current_state = self.SLEWING
```

Comment:
This isn't true. We don't start off slewing, nor should we. If you disagree, please explain why.

Reply:
Not moving is functionally equivalent to tracking the current position - ie slewing with a delta of 0.

Reply:
Ok, but then is_ready has undefined behaviour, because it's checking that the encoder is within the error of the target, but we haven't set a target. Thanks to magicbot, is_ready gets called before any slew commands are issued. Besides, this class should be impervious to any such call-order dependencies. The rest of this class makes some assumptions about preconditions when in the slewing state. If you are going to set that state at the beginning, please ensure that all those assumptions aren't violated so the rest of the class is robust.

Reply:
That is true.
Would you be happy for a call to reset the encoders and slew to position 0 in the on_enable() method?

Reply:
Sure. Whatever will ensure that the preconditions for the state are met such that no call to any other method will cause problems.

Reply:
Commit pushed that immediately engages position hold mode.

Reply:
Excellent.